packages/core/src/Asset/util/subtractTokenMaps.ts

```diff
@@ -0,0 +1,24 @@
+import { TokenMap } from '../../Cardano';
+import { util } from '../../util';
+
+/**
+ * Subtract asset quantities in order
+ */
+export const subtractTokenMaps = (totals: (TokenMap | undefined)[]): TokenMap | undefined => {
+  if (totals.length <= 0 || !util.isNotNil(totals[0])) return undefined;
+  const result: TokenMap = totals[0];
+  const rest: TokenMap[] = totals.slice(1).filter(util.isNotNil);
+  for (const assetTotals of rest) {
+    for (const [assetId, assetQuantity] of assetTotals.entries()) {
+      const total = result.get(assetId);
+      if (total) {
+        const diff = total - assetQuantity;
+        diff > 0 ? result.set(assetId, total) : result.delete(assetId);
```

Comment:
Given function name, I would expect this to return negative quantities too. Maybe let's create a separate function to check for negative quantities or to remove them.

```suggestion
        diff === 0n ? result.delete(assetId) : result.set(assetId, total);
```

Btw `subtractValueQuantities` can already return negative quantity of coin